### DISCUSSION TITLE
chemical potential in coupled phase field/mechanics problems

### DISCUSSION THREAD

**Discussion by jessecarterMOOSE** [idaholab/moose#17022] [Q&A Modules: Phase field] [answered]
In a coupled phase field/mechanics problem, should the chemical potential be defined as the derivative of the total system energy or just the chemical energy (as its name implies)? My understanding is that it's the total because for example in the CH equation you have df_chem/dc and dE_d/dc and so you supply the total energy to the `SplitCHParsed` kernel.

But for KKS problems, in the examples that I've seen, the CH kernels take just the chemical energies rather than the total. Is that because the free energy is differentiated with respect to the phase concentrations (rather than total), so other terms drop out? And as for the AC equation, it seems to matter if doing KHS or VTS interpolation? See [here](https://github.com/idaholab/moose/blob/next/modules/combined/examples/phase_field-mechanics/kks_mechanics_KHS.i) and [here](https://github.com/idaholab/moose/blob/next/modules/combined/examples/phase_field-mechanics/kks_mechanics_VTS.i).

Can someone help me understand how to properly use the chemical potential?

**Comment by dschwen** [comment]
You are correct, as far as that the derivative of the total system energy determines the chemical potential. For a KKS simulation you would have an elastic energy E which might depend on the global concentration variables. You need to apply the chain rule to compute the derivative of the elastic energy w.r.t the phase concentrations. In pure phase regions those derivatives would vanish for the inactive phases (as their phase concentration do not contribute to the global concentration because their switching function is zero). Within interfaces you'd have contributions from all phases. BUT, if I'm not mistaken, due to the linearity of the dependence of the global concentration on the phase concentrations you can basically take the derivative of the mechanical energy w.r.t the global concentration and multiply it with each phase's switching function to get the mechanical chemical potential contribution for that phase.

**Reply by jessecarterMOOSE** [reply]
Interesting @dschwen. What's the best way to go about this in MOOSE? I have a working example using the WBM formulation. Will the parsed materials system do the chain rule if I swap out my kernels for KKS kernels? Or do I need to make phase-specific elastic+chemical energies? That would make it more like a VTS interpolation rather than KHS, right?

Also I noticed that in @laagesen 's paper (https://doi.org/10.1016/j.commatsci.2017.08.005), equation 11 states the chemical potentials are taken using just the chemical energy? I didn't see any mention of mechanical energy in the original KKS paper.

**Reply by xueyang94** [reply]
The stiffness tensor and elastic strain can be interpolated to created global tensor and strain. The stress is then determined as a global stress. The elastic free energy is calculated using the KHS model. The chemical potential includes both the chemical energy and the elastic energy. This is published in a recent paper by Dr. Simon that might of help: [https://www.sciencedirect.com/science/article/pii/S0927025620302810]

**Reply by jessecarterMOOSE** [reply]
thanks @xueyang94 . That paper uses the Grand Potential formulation but I'm asking about KKS. Are they similar enough to apply the same logic? Right now I'm just trying to understand how to implement my system in MOOSE using WBM and KKS, both with KHS and VTS mechanical interpolation - so 4 different combinations. I was hoping to keep my comparison small but I will consider Grand Potential if it isn't much more work.

**Reply by xueyang94** [reply]
For the KKS and Grand Potential model (GPM), one can be used to derive another. They also use the same interfacial concentration partition condition, which is assuming the same chemical potential across the interface. So I would say they are similar enough that the same logic of handling the elastic energies can apply, just that GPM doesn't need to solve for phase concentration variables. The downside of GPM is that inverting the chemical and elastic energies from concentration-dependent to chemical potential-dependent can be a pain sometimes, so the energy form used in GPM is restricted. In terms of the WBM, since the interfacial concentration is defined in a different way, I am unsure if the same logic can be applied. Either way, interesting topic and good luck.

**Comment by dschwen** [comment, marked as answer]
I'm not sure such a consideration has been published before... I'm available as a co-author ;-)

**Reply by laagesen** [reply]
Hi @jessecarterMOOSE - in the work we have done up until now such as the paper that you mentioned, we have only rigorously considered order parameter dependent eigenstrains. So, in that case when you take the derivative of the elastic energy term w.r.t. concentration you get zero- meaning only the chemical energy contributes to chemical potential for both VTS and KHS schemes. The elastic strain energy term affects the evolution equation for the order parameter though.

If you wanted to consider a concentration-dependent eigenstrain, it's not totally clear to me off-hand whether you would include the elastic energy contribution when setting up the chemical potential constraint. It would really only be feasible for the VTS scheme, because in the VTS scheme there is a per-phase elastic energy defined. Whereas in the KHS scheme there is a global elastic energy defined and it isn't really separated out on a per-phase basis. Since more authors in the literature have used the KHS scheme, I would guess most have not included an elastic energy contribution to the chemical potential constraint equation.

**Reply by jessecarterMOOSE** [reply]
Thanks for your perspective @laagesen. I do indeed have a concentration-dependent eigenstrain so you understand the problem I'm trying to navigate. 

Looking at the [KKS KHS example](https://github.com/idaholab/moose/blob/next/modules/combined/examples/phase_field-mechanics/kks_mechanics_KHS.i) example, I see how the global elastic energy term gets added to the AC equation. I suppose I could do something similar for the CH equation. In addition, I could also define some DerivativeSumMaterials that combine my chemical and elastic energies for each phase, and use those as the free energies in the chemical potential constraint. The only problem is that I'm not sure what the "right" solution looks like.

I see in the [KKS VTS example](https://github.com/idaholab/moose/blob/next/modules/combined/examples/phase_field-mechanics/kks_mechanics_VTS.i) that you combined the chemical and elastic energies in each phase, but is that out of necessity or no? It seems you could still have only passed the chemical free energy of each phase to the chemical potential constraint kernel.

**Reply by laagesen** [reply]
Right, for the VTS example I could have passed the chemical energy directly to the constraint kernel. 

For your case of a concentration-dependent eigenstrain, most work I've seen uses the KHS scheme and only considers the chemical part of the free energy in the constraint. So I think you would be safe to do that, but I'm not sure it's fundamentally the right answer. What about trying it both ways and comparing the result for some simple test cases? You could even do something similar to our 2017 paper and the test cases we did there. It would be interesting to see which approach affected the interfacial energy the most; that is probably the right way to assess which approach works best, figure out which results in an interfacial energy that is closest to the values you set as input parameters. This could be the basis of a pretty interesting follow-on paper to the 2017 paper, actually...

**Reply by simopier** [reply]
@jessecarterMOOSE, I come late to the party, but I met the same challenge during my Ph.D. work (with which you are very familiar). I ended up including the elastic contribution in the "total" potential constraint equation of the KKS model. I remember doing a few test simulations to select the best approach but never considered publishing the results. If this problem is still relevant to you, I would be happy to chat with you about it and maybe get something worth publishing.
